src/org/pentaho/reporting/platform/plugin/JobManager.java

```diff
@@ -97,10 +97,6 @@ public JobManager( final boolean isSupportAsync, final long pollingIntervalMilli
     final IPentahoSession session = PentahoSessionHolder.getSession();
 
     Future<InputStream> future = executor.getFuture( uuid, session );
-    if ( future == null || !future.isDone() ) {
```

Comment:
Not checking future.isDone() we can potentially came into situation when we do request content future.get() and waiting for the competition. That will be a blocking operation. If we going to request content for 1st page calling /content end-point for content that is not ready it will be blocking request as it would be without async requests for first page. We may consider to request 'first page' in regular instead of async mode, asuming since 1st page is in cache - it is most probably will be available almost ready and we don't want to waste thread pool with such short-running tasks.